### PR TITLE
Feature/#509 change ed x course worker

### PR DESCRIPTION
**Pull request by rosabraatz**
#509 change edX worker
it now works with rss feed to collect course data

### CODE REVIEW

File: app/workers/edx_course_worker.rb [hunk 1]
```diff
@@ -20,92 +32,93 @@ def handle_response_data(response_data)
...
-    response_data['value']['items'].each do |course_element|
-      course = Course.find_by(provider_course_id: course_element['course:id'], mooc_provider_id: mooc_provider.id)
-      if course.nil?
-        course = Course.new
-      else
-        update_map[course.id] = true
-      end
+    response_data.each do |xml_doc|
+      xml_doc.xpath('//channel/item').each do |course_element|
+        course = Course.find_by(provider_course_id: course_element.xpath('course:id').text, mooc_provider_id: mooc_provider.id)
```
Comment (MrSerth) [inline app/workers/edx_course_worker.rb]: What about our special method `Course.get_course_id_by_mooc_provider_id_and_provider_course_id`? You don't have to use it, just if you like.

File: spec/workers/edx_course_worker_spec.rb [hunk 2]
```diff
@@ -57,21 +132,30 @@
...
-    json_course['course:staff'] = 'Person A, Person B'
-    json_course['course:subject'] = 'Topic'
-    edx_course_worker.handle_response_data json_course_data
-    course = Course.find_by(provider_course_id: json_course['course:id'], mooc_provider_id: mooc_provider.id)
-    expect(course.course_instructors).to eql json_course['course:staff']
-    expect(course.categories).to eql [json_course['course:subject']]
+  it 'assigns all instructors to course' do
+    xml_course = xml_course_data[0].xpath('//channel/item')
+    edx_course_worker.handle_response_data xml_course_data
+    course = Course.find_by(provider_course_id: xml_course.xpath('course:id').text, mooc_provider_id: mooc_provider.id)
```
Comment (MrSerth) [inline spec/workers/edx_course_worker_spec.rb]: `Course.get_course_id_by_mooc_provider_id_and_provider_course_id`?

File: spec/workers/edx_course_worker_spec.rb [hunk 3]
```diff
@@ -57,21 +132,30 @@
...
+    xml_course.xpath('course:instructors/course:staff').each do |staff|
+      expect(course.course_instructors).to include staff.xpath('staff:name').text
+    end
+  end
+
+  it 'assigns all subjects to course' do
+    xml_course = xml_course_data[0].xpath('//channel/item')
+    edx_course_worker.handle_response_data xml_course_data
+    course = Course.find_by(provider_course_id: xml_course.xpath('course:id').text, mooc_provider_id: mooc_provider.id)
```
Comment (MrSerth) [inline spec/workers/edx_course_worker_spec.rb]: `Course.get_course_id_by_mooc_provider_id_and_provider_course_id`?

File: spec/workers/edx_course_worker_spec.rb [hunk 4]
```diff
@@ -87,10 +171,11 @@
...
 
   it 'creates a xseries course track type' do
-    json_course = json_course_data['value']['items'][0]
-    json_course['course:xseries'] = '1'
-    edx_course_worker.handle_response_data json_course_data
-    course = Course.find_by(provider_course_id: json_course['course:id'], mooc_provider_id: mooc_provider.id)
+    xml_course = xml_course_data[0].xpath('//channel/item')
+    xml_course.xpath('course:xseries').first.content = '1'
+    edx_course_worker.handle_response_data xml_course_data
+    course = Course.find_by(provider_course_id: xml_course.xpath('course:id').text, mooc_provider_id: mooc_provider.id)
```
Comment (MrSerth) [inline spec/workers/edx_course_worker_spec.rb]: `Course.get_course_id_by_mooc_provider_id_and_provider_course_id`?

File: spec/workers/edx_course_worker_spec.rb [hunk 5]
```diff
@@ -106,10 +191,11 @@
...
 
   it 'creates a profed course track type' do
-    json_course = json_course_data['value']['items'][0]
-    json_course['course:profed'] = '1'
-    edx_course_worker.handle_response_data json_course_data
-    course = Course.find_by(provider_course_id: json_course['course:id'], mooc_provider_id: mooc_provider.id)
+    xml_course = xml_course_data[0].xpath('//channel/item')
+    xml_course.xpath('course:profed').first.content = '1'
+    edx_course_worker.handle_response_data xml_course_data
+    course = Course.find_by(provider_course_id: xml_course.xpath('course:id').text, mooc_provider_id: mooc_provider.id)
```
Comment (MrSerth) [inline spec/workers/edx_course_worker_spec.rb]: `Course.get_course_id_by_mooc_provider_id_and_provider_course_id`?

File: spec/workers/edx_course_worker_spec.rb [hunk 6]
```diff
@@ -57,21 +132,30 @@
...
 
   it 'creates a certificate course track type' do
-    json_course = json_course_data['value']['items'][0]
-    json_course['course:verified'] = '1'
-    edx_course_worker.handle_response_data json_course_data
-    course = Course.find_by(provider_course_id: json_course['course:id'], mooc_provider_id: mooc_provider.id)
+    xml_course = xml_course_data[0].xpath('//channel/item')
+    xml_course.xpath('course:verified').first.content = '1'
+    edx_course_worker.handle_response_data xml_course_data
+    course = Course.find_by(provider_course_id: xml_course.xpath('course:id').text, mooc_provider_id: mooc_provider.id)
```
Comment (MrSerth) [inline spec/workers/edx_course_worker_spec.rb]: `Course.get_course_id_by_mooc_provider_id_and_provider_course_id`?

**Comment by rosabraatz** [comment]
In order to use `Course.get_course_id_by_mooc_provider_id_and_provider_course_id `, I refactor a little bit. The method is now called `Course.get_course_by_mooc_provider_id_and_provider_course_id ` and returns the whole course instead of only the id.

**Comment by MrSerth** [comment]
Thanks for your refactoring (especially with the course method)! I think, it is ready for merging! :+1:

**Pull request merged**